### PR TITLE
HPCC-29687 MP test host ip:port parse fix

### PR DESCRIPTION
**Pull request by mckellyln**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [x] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

manual testing with the startn-mptests script

### CODE REVIEW

File: system/mp/test/mptest.cpp [hunk 1]
```diff
@@ -1451,6 +1451,22 @@ void runTest(const char *caption, const char *testname, IGroup* group, ICommunic
     comm->barrier();
 }
 
+bool getSocketEndpoint(char *inputStr, SocketEndpoint &ep)
```
Comment (jakesmith) [inline system/mp/test/mptest.cpp:1454]: can/should be static method?

File: system/mp/test/mptest.cpp [hunk 2]
```diff
@@ -1466,11 +1482,16 @@ bool createNodeList(IArrayOf<INode> &nodes, const char* hostfile, int my_port, r
         if ((max_ranks > 0) && ((i - 1) >= max_ranks))
             break;
 
-        int srtn = sscanf(line, "%s", hoststr);
-        if (srtn == 1 && line[0] != '#') {
-            INode* newNode = createINode(hoststr, my_port);
-            nodes.append(*newNode);
-            i++;
+        int srtn = sscanf(line, "%255s", hoststr);
```
Comment (jakesmith) [inline system/mp/test/mptest.cpp:1485]: not new perhaps, but feels like the:
```
if (line[0] != '#')
```

should come 1st before the sscanf

File: system/mp/test/mptest.cpp [hunk 3]
```diff
@@ -1466,11 +1482,16 @@ bool createNodeList(IArrayOf<INode> &nodes, const char* hostfile, int my_port, r
...
+        if (srtn == 1 && line[0] != '#')
```
Comment (jakesmith) [inline system/mp/test/mptest.cpp:1486]: why sscanf at all , could it be : 
```
if (line[0] != '#')
{
    SocketEndpoint hep;
    if (getSocketEndpoint(hoststr, hep))
    {
        INode *newNode = createINode(hep);
        nodes.append(*newNode);
        i++;
    }
}
```

File: system/mp/test/mptest.cpp [hunk 4]
```diff
@@ -1466,11 +1482,16 @@ bool createNodeList(IArrayOf<INode> &nodes, const char* hostfile, int my_port, r
...
+        {
+            SocketEndpoint hep;
+            if (getSocketEndpoint(hoststr, hep))
```
Comment (jakesmith) [inline system/mp/test/mptest.cpp:1489]: what if there is no :<port> - it used to handle hostname only and use passed in 'my_port',
now that argument is not used

**Review by jakesmith — COMMENTED** [review]
@mckellyln - please see comments.

File: system/mp/test/mptest.cpp (hunk 1 above)
Comment (mckellyln) [reply system/mp/test/mptest.cpp]: ok, sure

File: system/mp/test/mptest.cpp (hunk 3 above)
Comment (mckellyln) [reply system/mp/test/mptest.cpp]: I agree, changed.

File: system/mp/test/mptest.cpp (hunk 2 above)
Comment (mckellyln) [reply system/mp/test/mptest.cpp]: Yes, changed.

**Comment by mckellyln** [comment]
@jakesmith thanks, I have made changes, please re-review commit 2.

**Review by jakesmith — APPROVED** [review]
@mckellyln - looks good, 1 minor comment.

Tick box needed at top of PR.

Approving though.
Please address 1 comment and squash.

**Comment by mckellyln** [comment]
Squashed and check box updated.
@ghalliday ready for merge

**Comment by mckellyln** [comment]
Also deleted unused variable hoststr.

**Pull request merged**